<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>727</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>220</y>
      <width>80</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Check</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>281</width>
      <height>161</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700; text-decoration: underline; color:#aa0000;&quot;&gt;Welcome!!&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;This program can be used to:&lt;br /&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;1)&lt;/span&gt; Check whether a string of parentheses is balanced or not.&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;2) &lt;/span&gt;Check how many digits, letters, and symbols are in the string&lt;br /&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;3) &lt;/span&gt;Output entered string in reverse using a stack&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>220</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>input string:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="string1">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>220</y>
      <width>113</width>
      <height>24</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>90</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Balanced?</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="result">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>90</y>
      <width>113</width>
      <height>24</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>150</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Postfix expression:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="postfix">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>150</y>
      <width>113</width>
      <height>24</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="disclaimer">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>200</y>
      <width>271</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>! please DO NOT copy paste ; enter string manually !</string>
    </property>
   </widget>
   <widget class="QLabel" name="readonly">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>50</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>(READ ONLY)</string>
    </property>
    <property name="textFormat">
     <enum>Qt::AutoText</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>300</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <italic>false</italic>
      <bold>true</bold>
      <underline>true</underline>
     </font>
    </property>
    <property name="text">
     <string>FUNCTIONALITY 2</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>300</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <bold>true</bold>
      <underline>true</underline>
     </font>
    </property>
    <property name="text">
     <string>FUNCTIONALITY 3</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>0</y>
      <width>231</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <italic>true</italic>
      <strikeout>false</strikeout>
      <kerning>true</kerning>
      <fontweight>Black</fontweight>
     </font>
    </property>
    <property name="text">
     <string>Hashim Abdulla &amp; Mohamed Elgasim</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>330</y>
      <width>49</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLineEdit" name="input2">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>340</y>
      <width>113</width>
      <height>24</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="checkbutton2">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>340</y>
      <width>80</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Check</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>340</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>input string:</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>400</y>
      <width>101</width>
      <height>70</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700; color:#5500ff;&quot;&gt;Number of &lt;br /&gt;digits,letters, &lt;br /&gt;and symbols:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="answer2">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>400</y>
      <width>161</width>
      <height>111</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>440</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Reversed:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="reversebox">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>440</y>
      <width>181</width>
      <height>24</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>727</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuBalanced_checker">
    <property name="title">
     <string>Extra Credit Project</string>
    </property>
   </widget>
   <addaction name="menuBalanced_checker"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
